smile/niusb_interface.py

```diff
@@ -0,0 +1,166 @@
+#emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+#ex: set sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See the COPYING file distributed along with the smile package for the
+#   copyright and license terms.
+#
+#   NOTES: On Demand timing, the default, makes it so that the write() method
+#   doesn't return until all of the samples are written. This also means that
+#   we can get it to send different kinds of signals.
+#
+#   PLAN: enter is a write to VCC, Leave is a write to 0. Channel is designated
+#   by the state parameters. tasks must be setup before hand and passed into
+#   the state much like the servers.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+from state import State
+from event import event_time
+from clock import clock
+from ref import NotAvailable
+
+try:
+    from nidaqmx.task import Task
+    _got_nidaqmx = True
+except ImportError:
+    print("Unable to import pylsl!")
+    _got_nidaqmx = False
+
+
+_ni_tasks = {}
+
+
+def init_task(task_name="TASKING", min_val=0.0, max_val=1.0,
```

Comment:
Can we call the default task_name something else?

Reply:
ttl_pulse?

Reply:
Sure